sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileStorageResourceOptions.cs

```diff
@@ -47,22 +35,116 @@ public class ShareFileStorageResourceOptions
 
+        /// <summary>
+        /// Optional. Sets the Cache Control header which
+        /// specifies directives for caching mechanisms.
+        ///
+        /// By default preserves the Cache Control from the source.
+        ///
+        /// Applies to upload and copy transfers.
+        /// </summary>
+        public DataTransferProperty<string> CacheControl { get; set; }
+
+        /// <summary>
+        /// Optional. Sets the Content Disposition header which
+        /// conveys additional information about how to process the response
+        /// payload, and also can be used to attach additional metadata.  For
+        /// example, if set to attachment, it indicates that the user-agent
+        /// should not display the response, but instead show a Save As dialog
+        /// with a filename other than the blob name specified.
+        ///
+        /// By default preserves the Content Disposition from the source.
+        ///
+        /// Applies to upload and copy transfers.
+        /// </summary>
+        public DataTransferProperty<string> ContentDisposition { get; set; }
+
+        /// <summary>
+        /// Optional. Sets the Content Encoding header which
+        /// specifies which content encodings have been applied to the blob.
+        /// This value is returned to the client when the Get Blob operation
+        /// is performed on the blob resource. The client can use this value
+        /// when returned to decode the blob content.
+        ///
+        /// By default preserves the Content Encoding from the source.
+        ///
+        /// Applies to upload and copy transfers.
+        /// </summary>
+        public DataTransferProperty<string[]> ContentEncoding { get; set; }
+
+        /// <summary>
+        /// Optional. Sets the Content Language header which
+        /// specifies the natural languages used by this resource.
+        ///
+        /// By default preserves the Content Language from the source.
+        ///
+        /// Applies to upload and copy transfers.
+        /// </summary>
+        public DataTransferProperty<string[]> ContentLanguage { get; set; }
+
+        /// <summary>
+        /// Optional. Sets the Content Type header which
+        /// specifies the MIME content type of the blob.
+        ///
+        /// By default preserves the Content Type from the source.
+        ///
+        /// Applies to upload and copy transfers.
+        /// </summary>
+        public DataTransferProperty<string> ContentType { get; set; }
+
+        /// <summary>
+        /// The file system attributes for this file.
+        ///
+        /// By default preserves the File Attributes from the source.
+        /// </summary>
+        public DataTransferProperty<NtfsFileAttributes?> FileAttributes { get; set; }
+
+        /// <summary>
+        /// The key of the file permission.
+        /// </summary>
+        public string FilePermissionKey { get; set; }
```

Comment:
This is probably going to be the next PR but how do we plan on handling permissions and permission key? Are they going to be `DataTransferProperty` or will we introduce a `bool` for whether we preserve or not? We probably also need to think about the fact that I think only one or the other of these can be specified and if there is a permission key, there shouldn't be permissions?

Also noticed some things in this PR may end up changing w.r.t. permission key since it's not just a simple copy of the source key to the destination. We'll need to create a new key in the destination and then use that, so it won't match the source anymore. For now tests and such assume they will match.

Reply:
Yeah, I think I might go down the route of either introducing a `bool` or a regular `DataTransferProperty` (without the `<T>`). I don't see the point add the option to manually set the value of the file permission key, if we are going to do GetProperties / GetDirectoriesAndFiles which would give us the permission key anyways.

And I'm on the same page on the second part of this too, the permission wouldn't match the source any more.